corehq/warehouse/models/facts.py

```diff
@@ -0,0 +1,108 @@
+from django.db import models, transaction, connections
+
+from corehq.warehouse.const import (
+    APP_STATUS_FACT_SLUG,
+    FORM_FACT_SLUG,
+    USER_DIM_SLUG,
+    DOMAIN_DIM_SLUG,
+    FORM_STAGING_SLUG,
+    SYNCLOG_STAGING_SLUG,
+)
+
+from .dimensions import UserDim, DomainDim
+from corehq.form_processor.models import XFormInstanceSQL
+from corehq.sql_db.routers import db_for_read_write
+from corehq.util.test_utils import unit_testing_only
+from corehq.warehouse.etl import CustomSQLETLMixin
+from corehq.warehouse.models.shared import WarehouseTableMixin
+
+
+class BaseFact(models.Model, WarehouseTableMixin):
+
+    @classmethod
+    @transaction.atomic
+    def commit(cls, start_datetime, end_datetime):
+        cls.load(start_datetime, end_datetime)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    @unit_testing_only
+    def clear_records(cls):
+        database = db_for_read_write(cls)
+        with connections[database].cursor() as cursor:
+            cursor.execute("TRUNCATE {}".format(cls._meta.db_table))
```

Comment:
maybe we can pull this out into its own function? seems like it's used in a couple places